internal/telemetry/client.go

```diff
@@ -131,87 +138,91 @@ type Client struct {
 	newMetrics bool
+
+	// logLock gaurds the logging field
+	logLock sync.RWMutex
+	// Logging allows us to toggle on agentless in the case where
+	// there are issues with sending telemetry to the agent
+	Logging bool
 }
 
 func (c *Client) log(msg string, args ...interface{}) {
-	if c.Logger == nil {
+	// we don't log if the client is temporarily using agentless
+	// to avoid spamming the user with instrumentation telemetry error messages
+	c.logLock.RLock()
+	defer c.logLock.RUnlock()
+	if !c.Logging {
 		return
 	}
-	c.Logger.Printf(msg, args...)
+	log.Warn(fmt.Sprintf(LogPrefix+msg, args...))
+}
+
+// logging is used to turn logging on/off
+func (c *Client) logging(logging bool) {
+	c.logLock.Lock()
+	defer c.logLock.Unlock()
+	c.Logging = logging
+}
+
+// Started returns whether the client has started or not
+func (c *Client) Started() bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	return c.started
 }
 
 // Start registers that the app has begun running with the given integrations
 // and configuration.
-func (c *Client) Start(integrations []Integration, configuration []Configuration) {
+// TODO: implement passing in error information about tracer start
+func (c *Client) Start(configuration []Configuration) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
-	if c.Disabled || !internal.BoolEnv("DD_INSTRUMENTATION_TELEMETRY_ENABLED", true) {
+	if !internal.BoolEnv("DD_INSTRUMENTATION_TELEMETRY_ENABLED", true) || c.Disabled {
 		return
 	}
 	if c.started {
 		return
 	}
-	c.debug = internal.BoolEnv("DD_INSTRUMENTATION_TELEMETRY_DEBUG", c.debug)
-
 	c.started = true
 
 	// XXX: Should we let metrics persist between starting and stopping?
 	c.metrics = make(map[string]*metric)
+	c.applyDefaultOps()
 
 	payload := &AppStarted{
-		Integrations:  append([]Integration{}, integrations...),
 		Configuration: append([]Configuration{}, configuration...),
-	}
-	deps, ok := debug.ReadBuildInfo()
-	if ok {
-		for _, dep := range deps.Deps {
-			payload.Dependencies = append(payload.Dependencies,
-				Dependency{
-					Name:    dep.Path,
-					Version: dep.Version,
-				},
-			)
-		}
+		Products: Products{
+			AppSec: ProductDetails{
+				Version: version.Tag,
+				Enabled: appsec.Enabled(),
+			},
+		},
 	}
 
-	// configEnvFallback returns the value of environment variable with the
-	// given key if def == ""
-	configEnvFallback := func(key, def string) string {
-		if def != "" {
-			return def
-		}
-		return os.Getenv(key)
-	}
-	c.Service = configEnvFallback("DD_SERVICE", c.Service)
-	if len(c.Service) == 0 {
-		if name := globalconfig.ServiceName(); len(name) != 0 {
-			c.Service = name
-		} else {
-			// I think service *has* to be something?
-			c.Service = "unnamed-go-service"
+	appStarted := c.newRequest(RequestTypeAppStarted)
+	appStarted.Body.Payload = payload
+	c.scheduleSubmit(appStarted)
+
+	if internal.BoolEnv("DD_TELEMETRY_DEPENDENCY_COLLECTION_ENABLED", true) || c.CollectDependencies {
+		depPayload := Dependencies{[]Dependency{}}
```

Comment:
```suggestion
		var depPayload Dependencies
```
nit: I think(?) this should be equivalent